docs/sdk-architecture.md

```diff
@@ -0,0 +1,862 @@
+# C2PA Rust SDK Architecture Overview
+
+This document provides a comprehensive overview of the core SDK architecture in the `c2pa-rs` project. The SDK implements the [C2PA technical specification](https://c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html) and enables applications to create, sign, embed, and validate C2PA manifests in various asset types.
+
+## Table of Contents
+
+- [Overview](#overview)
+- [Core Architecture](#core-architecture)
+- [Major Components](#major-components)
+  - [Public API Layer](#public-api-layer)
+  - [Manifest Store & Claims](#manifest-store--claims)
+  - [Assertions](#assertions)
+  - [Ingredients](#ingredients)
+  - [Cryptography & Signing](#cryptography--signing)
+  - [Asset Handlers](#asset-handlers)
+  - [JUMBF Structure](#jumbf-structure)
+  - [Identity & Validation](#identity--validation)
+  - [Resource Management](#resource-management)
+- [Data Flow](#data-flow)
+- [Key Design Patterns](#key-design-patterns)
+
+## Overview
+
+The C2PA Rust SDK provides a comprehensive implementation for working with Content Credentials (C2PA manifests). It supports:
+
+- **Reading**: Parse and validate C2PA manifests from signed assets
+- **Writing**: Create and sign new C2PA manifests
+- **Validation**: Verify cryptographic signatures and trust chains
+- **Multiple formats**: JPEG, PNG, TIFF, GIF, MP3, MP4, PDF, SVG, and more
+- **Identity assertions**: Support for CAWG identity specifications
+- **Async/sync APIs**: Full support for both synchronous and asynchronous operations
+
+## Core Architecture
+
+The SDK follows a layered architecture design:
+
+```mermaid
+graph TB
+    subgraph "Public API Layer"
+        Reader["Reader<br/>(Read & Validate)"]
+        Builder["Builder<br/>(Create & Sign)"]
```

Comment:
Signer is currently part of the top level API, (as is Settings, but, of course more to talk about there)